numalogic/udfs/tools.py

```diff
@@ -107,6 +121,22 @@ def _load_model(
             payload.composite_keys,
             payload.metrics,
         )
-        return None
+        return None, payload
     else:
-        return artifact
+        _LOGGER.info(
+            "%s - Loaded Model. Source: %s , version: %s, Keys: %s, %s",
+            payload.uuid,
+            artifact.extras.get("source"),
+            artifact.extras.get("version"),
+            skeys,
+            dkeys,
+        )
+        if artifact.metadata and "artifact_versions" in artifact.metadata:
```

Comment:
shouldn't this be updated only when there is not artifact_versions in the payload metadata?

Reply:
Even though the artifact metadata is same foe all 3 set of models but yes I can add another if to remove redundant update of artifact_version.
Good catch!